Rubberduck.Core/UI/SelectionChangeService.cs

```diff
@@ -33,47 +34,60 @@ public SelectionChangeService(IVBE vbe, IParseCoordinator parser)
+        private void OnVbeSelectionChanged(object sender, EventArgs e)
         {
-            if (e.CodePane == null || e.CodePane.IsWrappingNullReference)
-            {
-                return;
-            }
-
             new Task(() =>
             {
-                var eventArgs = new DeclarationChangedEventArgs(e.CodePane, _parser.State.FindSelectedDeclaration(e.CodePane));
-                DispatchSelectedDeclaration(eventArgs);
+                using (var active = _vbe.ActiveCodePane)
+                {
+                    if (active == null)
+                    {
+                        return;
+                    }
+                    var eventArgs = new DeclarationChangedEventArgs(_parser.State.FindSelectedDeclaration(active));
+                    DispatchSelectedDeclaration(eventArgs);
+                }
             }).Start();
         }
 
         private void OnVbeFocusChanged(object sender, WindowChangedEventArgs e)
         {
             if (e.EventType == FocusType.GotFocus)
             {
-                switch (e.Window.Type)
+                switch (e.Hwnd.ToWindowType())
                 {
-                    case WindowKind.Designer:
-                        //Designer or control on designer form selected.
-                        if (e.Window == null || e.Window.IsWrappingNullReference || e.Window.Type != WindowKind.Designer)
+                    case WindowType.DesignerWindow:
+                        new Task(() =>
                         {
-                            return;
-                        }
-                        new Task(() => DispatchSelectedDesignerDeclaration(_vbe.SelectedVBComponent)).Start();                  
+                            using (var component = _vbe.SelectedVBComponent)
+                            {
+                                DispatchSelectedDesignerDeclaration(component);
+                            }                           
+                        }).Start();                  
                         break;
-                    case WindowKind.CodeWindow:
+                    case WindowType.CodePane:
                         //Caret changed in a code pane.
-                        if (e.CodePane != null && !e.CodePane.IsWrappingNullReference)
+                        new Task(() =>
                         {
-                            new Task(() => DispatchSelectedDeclaration(new DeclarationChangedEventArgs(e.CodePane, _parser.State.FindSelectedDeclaration(e.CodePane)))).Start(); 
-                        }
+                            using (var pane = VBENativeServices.GetCodePaneFromHwnd(e.Hwnd))
+                            {
+                                DispatchSelectedDeclaration(
+                                    new DeclarationChangedEventArgs(_parser.State.FindSelectedDeclaration(pane)));
+                            }
+                        }).Start();
```

Comment:
why not `Task.Run(() => ...)`? That sidesteps the explicit `Start` and constructor